src/components/SnippetArea.jsx

```diff
@@ -18,13 +20,18 @@ const snippetEditorModes = {
   python3: 'python',
 };
 
-const SnippetArea = ({ contents, isDialogOpen, onTitleChanged, onSnippetChanged, readOnly, switchReadOnlyMode, snippetLanguage, toggleConfirmLockDialogVisibility }) => {
+
+const SnippetArea = ({ contents, isDialogOpen, onTitleChanged, onSnippetChanged,
```

Comment:
The amount of props we're passing down is getting ridiculous (not wrong, just annoying).